### DISCUSSION TITLE
Lazy::when can be overwritten by include()

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#39] [Q&A]
Hello there,

I am just a bit confused at the moment.

If I call

```php
public static function fromModel(User $user): self
{
    return new self(
        profile: Lazy::when(fn () => 1 === 2, fn () => "Test"),
    );
}
```

The result always contains the `profile`, if the User is called like this:

```php
UserData::from(User::first())->include('profile')
```

so the `include('profile')` determines the output - if the prop is set there, the result of the closure in `Lazy::when` has no meaning.

So my question is: Is it possible to exclude properties, even if they are manually included by a query string or `include`?

**Comment by rubenvanassche** [comment]
Hi you're right, this is a bit strange and it should be fixed in version 1.2.4!
